### PR TITLE
Update Rust to 1.77.1, min supported Rust to 1.74

### PR DESCRIPTION
**Pull request by mhammond**

### CODE REVIEW

File: docs/policies/rust-versions.md [hunk 1]
```diff
@@ -60,3 +51,19 @@ In the interests of avoiding redundant information which will inevitably
...
+* In [our circle CI integration](https://github.com/mozilla/uniffi-rs/blob/main/.circleci/config.yml)
+  you will find a number of docker references similar to `cimg/rust:1.XX` - all of these
+  should be updated; while some tasks, such as publishing the docs, are largely indepdenent of
+  the requirements in this policy, we should keep them all consistent where we can.
+
+## Updating the "Minimum Supported Version
+
+* In [our circle CI integration](https://github.com/mozilla/uniffi-rs/blob/main/.circleci/config.yml)
+  you will find a task `prepare-rust-min-version` which specifies this version via executing
+  `rustup update` with the version. This is the version to update.
```
Comment (bendk) [inline docs/policies/rust-versions.md:69]: My future self thanks you for this one.

**Review by bendk — APPROVED** [review]
Looks good, although I think the trybuild tests need updating

Hopefully this will fix the `package `cargo-platform v0.1.8` cannot be built because it requires rustc 1.73 or newer, while the currently active rustc version is 1.72.0` error that I ran into when publishing `0.27.0`.  Did you hit that when you published `0.27.1`?

**Comment by mhammond** [comment]
> Hopefully this will fix the `package `cargo-platform v0.1.8` cannot be built because it requires rustc 1.73 or newer, while the currently active rustc version is 1.72.0` error that I ran into when publishing `0.27.0`. Did you hit that when you published `0.27.1`?

I did - that's my motivation for this PR :)

**Comment by mhammond** [comment]
There's no urgency here so I'll wait for @badboy to chime in

**Review by badboy — APPROVED** [review]

**Pull request merged**